### DISCUSSION TITLE
Handling incoming dynamic link when app was not yet installed on iOS

### DISCUSSION THREAD

**Discussion by kickbk** [invertase/react-native-firebase#6275] [Q&A] [answered]
Most of the literature I find online, including one on react-native-firebase, explains how one handles an incoming dynamic link when the app was already installed.
However, when the app was not yet installed, it will redirect to the app stores, install the app, open it, and pass a different link to the app in case of iOS, something like `com.the.app://google/link/?deep_link_id=https://somewebfallback.com/page.html?vars=xxxxxxxxx&anothervar=xxxx&match_type=unique&utm_medium=dynamic_link&request_ip_version=IP_V6&utm_source=firebase&match_message=Link is uniquely matched for this device.`

On Android it will already "open" the dynamic link and pass the value which is above under `deep_link_id`. so this question is about iOS specifically.

The params we care about are contained in `deep_link_id`.

I created a front + background listeners as follows:
```js
const handleDeepLink = useCallback(
  async (link: FirebaseDynamicLinksTypes.DynamicLink | null) => {
     // Is link going to be the same as in deep_link_id if the app was just installed via the iOS app store?
    ...
  },
  []
);

useEffect(() => {
  const unsubscribe = dynamicLinks().onLink(handleDeepLink);
  return unsubscribe;
}, [handleDeepLink]);
useEffect(() => {
  dynamicLinks().getInitialLink().then(handleDeepLink);
}, [handleDeepLink]);
```

You can see my question in the code above: If we were redirected to the app store, installed the app and it opened it with the url containing `deep_link_id` - will the `url` destructured from `link` be the full one I wrote above or the embedded one under `deep_link_id`?
Do you see what I am asking? Is react-native-firebase handling the part where the app is installed and opened by the app store for the first time cleaning up the left overs, or do we need to do it on our own?
I hope I'm being clear. It's kinda tough to explain.

Until recently I handled it without react-native-firebase and first parsed the link, then did 
```js
const parsedURL = Linking.parse(url);
if (parsedURL.queryParams?.deep_link_id) {
  ...
}
```
 So maybe I can avoid this with react-native-firebase?

**Comment by mikehardy** [comment, marked as answer]
I'm not sure I understand, but I think the resolveLink method will handle anything you throw at it, removing consideration of "what's my link like?" - I wrote it for that purpose https://rnfirebase.io/reference/dynamic-links#resolveLink

If I do understand, then my understanding of deep links indicates you should get the actual deep link

I *think* you can get a flowchart with details via methods described here: https://firebase.google.com/docs/dynamic-links/debug
(not totally related but somewhat related, v14.10.x implemented the `performDiagnostics` method for iOS, though I don't think it addresses your exact question)

Thinking less speculatively, why guess, try it, does it not work for you?
If not, then empirical data rules: what do the links actually look like when you test it?

**Comment by kickbk** [comment]
Thanks Mike! I definitely need to test it out, but it's nice to know that in theory it should work as I expect it to.
